README.md

```diff
@@ -91,6 +103,18 @@ docker-ci-deploy --tag-version 1.2.3 --tag-latest my-image
 ```
 This will result in the tags `my-image:1.2.3` and `my-image:latest` being pushed.
 
+#### Semantic version tags
+```
+docker-ci-deploy --tag alpine --tag-version 1.2.3 --tag-semver my-image
+```
+This will result in the tags `my-image:1.2.3-alpine`, `my-image:1.2-alpine`, and `my-image:1-alpine` being created and pushed. If part of the version is already present in the start of a tag, it will not be added. For example, in the above example if `--tag 1.2-alpine` were provided, the image would still be tagged with `1.2.3-alpine`, not `1.2.3-1.2-alpine`.
+
+This works by stripping pieces from the front of the version string using the regex `[.-]?\w+$`. This means that version strings with basic development/pre-release tags are also supported. For example, a tag such as `5.5.0-beta0` will produce the tags/tag prefixes `5.5.0-beta0`, `5.5.0`, `5.5`, and `5`.
```

Comment:
This is a bit complicated for no real reason :( basically, docker-ci-deploy doesn't know what a pre-release or stable release is. All I'm trying to say here is that the version string generator is just a bit more flexible than only taking numbers separated by `.`.

If you wanted to actually do a development release it'd probably be best to tag it explicitly to a git SHA or something and not use `--tag-semver`.

Let me try to reword this...